### PR TITLE
Sending invitations to respondents

### PR DESCRIPTION
**Pull request by maartje**
Use the public API to invite respondents.

Changes in this PR:
* Implement functionality to send an email invitation to a list of respondents

### CODE REVIEW

File: Library/Models/InvitationBatch.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+        /// <summary>
+        /// The name of the column that contains the email address
+        /// </summary>
+        public string EmailColumnName { get; set; }
+
+        /// <summary>
+        /// The id of the sample record
+        /// </summary>
+        public List<string> RespondentKeys { get; set; }
```
Comment (neilboyd) [inline Library/Models/InvitationBatch.cs]: Should be IEnumerable

File: Library/Services/Implementation/NfieldSurveyInviteRespondentsService.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+            var uri = SurveyInviteRespondentsUrl(surveyId);
+            var batchWithFilter = new InvitationBatchWithFilter()
+            {
+                EmailColumnName = batch.EmailColumnName,
+                InvitationTemplateId = batch.InvitationTemplateId,
+                Name = batch.Name,
+                ScheduledFor = batch.ScheduledFor
+            };
+            batchWithFilter.Filters = new[]
```
Comment (neilboyd) [inline Library/Services/Implementation/NfieldSurveyInviteRespondentsService.cs]: setting Filters can be inlined in the initializer

File: Library/Models/InvitationBatchWithFilter.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+using System;
+using System.Collections.Generic;
+
+namespace Nfield.Models
+{
+    /// <summary>
+    /// Class representing a batch of invitations
+    /// </summary>
+    public class InvitationBatchWithFilter
```
Comment (krakie) [inline Library/Models/InvitationBatchWithFilter.cs]: can be made internal

File: Library/Models/SampleFilter.cs [hunk 4]
```diff
@@ -20,7 +20,7 @@ namespace Nfield.Models
     /// <summary>
     /// class to define sample filters
     /// </summary>
-    internal class SampleFilter
+    public class SampleFilter
```
Comment (krakie) [inline Library/Models/SampleFilter.cs]: can be made internal

**Pull request merged**